src/firefly/java/edu/caltech/ipac/util/AppProperties.java

```diff
@@ -221,14 +222,27 @@ public static String getProperty(String key, String def, Properties overridePDB)
           try {
                retval= System.getProperty(key, def);
           } catch (Exception e) { /*do nothing*/ }
-          if (retval==def) retval= _mainProperties.getProperty(key, def);
+          if (retval==def) retval= mainProperties.getProperty(key, def);
       }
       else {
           retval= overridePDB.getProperty(key, def);
       }
       return retval;
   }
 
+    public static Map<String,String> getWithStartingMatch(String startOfKey) {
+        var retMap= new HashMap<String,String>();
+        for(Map.Entry<Object,Object>  entry : System.getProperties().entrySet()) {
+            var key= entry.getKey().toString();
+            if (key.startsWith(startOfKey)) retMap.put(key,(String)entry.getValue());
+        }
```

Comment:
This threw me off initially. I thought AppProperties was supposed to encapsulate this, but apparently it's implemented in such an odd way that you have to repeat it here. We should refactor AppProperties at some point.

Reply:
I don't understand your concern here. This is simple of function to return a maps of all properties that starts with a string.

Reply:
Oh, I was referring to [this](https://github.com/Caltech-IPAC/firefly/blob/1d408ca5c030739095e19e6514633daf326e29e4/src/firefly/java/edu/caltech/ipac/util/AppProperties.java#L209).
The logic of how to resolve a property is repeated here.  It's not a big deal.  There are other part of this class that can use some cleanup as well.

Reply:
yea, we can clean it up someday. I just removed an underscore. I don't think this is very high priority.